src/sidebar/helpers/mention-suggestions.ts

```diff
@@ -44,7 +51,20 @@ export function combineUsersForMentions(
       addedUserIds.add(userid);
       return !usernameAlreadyAdded;
     })
-    .sort((a, b) => a.username.localeCompare(b.username));
+    .sort((a, b) => {
+      if (mentionMode === 'username') {
+        return a.username.localeCompare(b.username);
+      }
+
+      const displayNameA = a.displayName ?? '';
+      const displayNameB = b.displayName ?? '';
+
+      // For display-name mentions, sort by display name first, then by username
+      return (
+        displayNameA.localeCompare(displayNameB) ||
```

Comment:
I think we should use a case and accent-insensitive comparison for display names. Otherwise a list with names in a mix of cases could have ordering that is quite surprising to someone who is not a software engineer. This can be done using `{ sensitivity: "base" }` in the `localeCompare` options.